src/NServiceBus.AcceptanceTests/Routing/When_doing_content_based_routing_with_commands.cs

```diff
@@ -0,0 +1,153 @@
+                        });
+                    c.GetSettings().GetOrCreate<DistributionPolicy>()
+                        .SetDistributionStrategy(new ContentBasedRoutingStrategy(ReceiverEndpoint));
+                });
+            }
+
+            class ContentBasedRoutingStrategy : DistributionStrategy
+            {
+                public ContentBasedRoutingStrategy(string endpoint) : base(endpoint, DistributionStrategyScope.Send)
+                {
+                }
+
+                public override string SelectReceiver(string[] receiverAddresses)
+                {
+                    throw new NotImplementedException(); // should never be called
+                }
+
+                public override Task<string> SelectReceiver(string[] receiverAddresses, IOutgoingContext outgoingContext)
+                {
+                    var sendContext = outgoingContext as IOutgoingSendContext;
+                    var message = sendContext?.Message.Instance as MyCommand;
+                    if (message != null)
+                    {
+                        return Task.FromResult(receiverAddresses.Single(a => a.Contains(message.Instance)));
+                    }
+                    throw new InvalidOperationException("Unable to route!");
+                }
+            }
+        }
+
+        public class Receiver1 : EndpointConfigurationBuilder
+        {
+            public Receiver1()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.MakeInstanceUniquelyAddressable(Discriminator1);
+                }).CustomEndpointName(ReceiverEndpoint);
+            }
+
+            public class MyCommandHandler : IHandleMessages<MyCommand>
+            {
+                public Context Context { get; set; }
+
+                public Task Handle(MyCommand message, IMessageHandlerContext context)
+                {
+                    Interlocked.Increment(ref Context.MessageDeliveredReceiver1);
+                    return Task.FromResult(0);
+                }
+            }
+        }
+
+        public class Receiver2 : EndpointConfigurationBuilder
+        {
+            public Receiver2()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.MakeInstanceUniquelyAddressable(Discriminator2);
+                }).CustomEndpointName(ReceiverEndpoint);
+            }
+
+            public class MyCommandHandler : IHandleMessages<MyCommand>
+            {
+                public Context Context { get; set; }
+
+                public Task Handle(MyCommand message, IMessageHandlerContext context)
+                {
+                    Interlocked.Increment(ref Context.MessageDeliveredReceiver2);
+                    return Task.FromResult(0);
+                }
+            }
+        }
+
+        public class MyCommand : ICommand
+        {
+            public string Instance { get; set; } 
+        }
+    }
+}
```

Comment:
Yeah I saw that but I explicitly wanted two endpoints so that I can have dedicated handlers to verify that stuff is properly routed. If I use the same Endpoint I can make them uniquely addressable but can't have two handlers. Then the handler impl would need to determine on which addressable instance it was received.